source/agora/consensus/Fee.d

```diff
@@ -427,17 +457,19 @@ public class FeeManager
             foreach (input; tx.inputs)
             {
                 UTXO utxo;
-                assert(peekUTXO(input.utxo, utxo));
+                if (!peekUTXO(input.utxo, utxo))
+                    return "Unable to find input for utxo";
                 tot_in.mustAdd(utxo.output.value);
             }
 
-            assert(tx.getSumOutput(tot_out), "Not validated block in" ~
-                "getTXSetFees");
+            if (!tx.getSumOutput(tot_out))
+                return "Transaction output value is invalid";
```

Comment:
If those get triggered, isn't it logic error ?

Reply:
this function can get (and is called) with partially verified transactions (transactions that got relayed to us by an attacker through putTransaction). It can very well happen that those transactions didn't have their input in the UTXO or their total sum doesn't add up, so we shouldn't assert